api/app/GraphQL/Validators/ChangePoolClosingDateValidator.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\GraphQL\Validators;
+
+use App\Models\Pool;
+use Database\Helpers\ApiErrorEnums;
+use Illuminate\Validation\Rule;
+use Nuwave\Lighthouse\Validation\Validator;
+
+final class ChangePoolClosingDateValidator extends Validator
+{
+    /**
+     * Return the validation rules.
+     *
+     * @return array<string, array<mixed>>
+     */
+    public function rules(): array
+    {
+
+        $id = $this->arg('id');
+        $pool = Pool::find($id);
+
+        return [
+            'id' => ['uuid', 'required'],
+            'newClosingDate' => [
+                'after:today',
+                Rule::when(! is_null($pool) && $pool->closing_date, ['after_or_equal:'.$pool?->closing_date]),
+            ],
```

Comment:
I know `new_closing_date` and `newClosingDate` already existed but I find them confusing as there is no such field and it becomes a chain of 
closing_date -> new_closing_date -> closing_date
for the validation and update process